routers/admin/auths.go

```diff
@@ -152,6 +159,16 @@ func parseOAuth2Config(form auth.AuthenticationForm) *models.OAuth2Config {
 	}
 }
 
+func parseSSPIConfig(form auth.AuthenticationForm) *models.SSPIConfig {
+	return &models.SSPIConfig{
+		AutoCreateUsers:      form.SSPIAutoCreateUsers,
+		AutoActivateUsers:    form.SSPIAutoActivateUsers,
+		StripDomainNames:     form.SSPIStripDomainNames,
+		SeparatorReplacement: form.SSPISeparatorReplacement,
+		DefaultLanguage:      form.SSPIDefaultLanguage,
```

Comment:
I couldn't find any parameter validation for this (may be I missed it, there's so many files in this PR).

Reply:
I missed that. Was trying to follow example by SMTP config, eg:
https://github.com/go-gitea/gitea/blob/9f67df8cd75e7bceb3ee8077ccc4c84b455dfdf6/routers/admin/auths.go#L129

But I don't fully understand how other authentication sources validate parameters.

Can you hint on the idiomatic way (for the project) to validate those.

Reply:
> Can you hint on the idiomatic way (for the project) to validate those, to avoid inventing my own.

This is one simple example:

https://github.com/go-gitea/gitea/blob/108bed202349532d22f55c51a794ad48ae662e17/routers/repo/issue.go#L562-L565

Then here we use a marker to highlight the problem field:

https://github.com/go-gitea/gitea/blob/108bed202349532d22f55c51a794ad48ae662e17/routers/admin/users.go#L110-L114

(the marker is referenced in the template)

Reply:
Note: I'd use a regexp to enforce the format of the field:
```
ok, err := regexp.MatchString(`^[a-z]{2}-[A-Z]{2}$`, value)
```

Reply:
Validation introduced with commit https://github.com/go-gitea/gitea/pull/8463/commits/7e24021f7d0f2ba089f880a780478cc7ade794c6 (ammended with https://github.com/go-gitea/gitea/pull/8463/commits/6c9273a1c7927819d2cd83122cdda0843973da80)

Reply:
Added validation with commit https://github.com/go-gitea/gitea/pull/8463/commits/0e37157668c533d2695b6f5b28707046b58f3775